mlir/lib/Dialect/Vector/Transforms/VectorEmulateNarrowType.cpp

```diff
@@ -149,6 +152,48 @@ static Value insertSubvectorInto(RewriterBase &rewriter, Location loc,
                                                        dest, offsets, strides);
 }
 
+/// Extracts `lengthSubvec` elements from `srcVec` into `destVec` starting at
+/// the offset specified by `srcOffsetVar`. Use this function when
+/// `srcOffsetVar` is not a constant, making it impossible to use
+/// vector.extract_strided_slice, as it requires constant offsets.
+static Value dynamicallyExtractSubVector(RewriterBase &rewriter, Location loc,
+                                         TypedValue<VectorType> srcVec,
+                                         Value destVec,
+                                         OpFoldResult srcOffsetVar,
+                                         int64_t lengthSubvec) {
```

Comment:
[nit] `srcVec` -> `source` (we know it's a vector)
[nit] `destVec` -> `dest` (same)
[nit] `srcOffsetVar` -> `offsets` (we know it's a _Var_iable and, at least to me, offsets are tied to ExtractOp rather than the source itself)
[nit] `lengthSubvec` -> `numElementsToExtract`? (based on your other comments, feels more accurate)